### DISCUSSION TITLE
Create deployment using REST API

### DISCUSSION THREAD

**Discussion by Aarush-Acharya** [vercel/community#3550] [Help] [answered]
I want to create a deployment in an already existing project using the REST API deployment endpoint 

Cannot figure out how to do it

if I do not provide the project info in the body then it creates a new project and starts deploying there 

I want it to create a new deployment in an existing project how to do that

currently doing this 

<img width="1423" alt="Screenshot 2023-08-09 at 4 17 20 PM" src="https://github.com/vercel/community/assets/92685647/ecde732d-e2ce-433d-bef3-5e9a729c53fb">

Auth token has been taken care of

**Comment by amyegan** [comment, marked as answer]
Hey @Aarush-Acharya. You can [create a new deployment](https://vercel.com/docs/rest-api/endpoints#create-a-new-deployment) via the `/v13/deployments` endpoint. 

To create a new deploy from a GitHub repo, you could use a body like
```
{
  "gitSource": {
		"ref" : "<BRANCH-NAME>",
		"repoId": "<GITHUB-REPO-ID>",
		"type": "github",
		"sha": "<COMMIT-SHA>"
	},
  "name": "<PROJECT-NAME>"
}
```

**Comment by Aarush-Acharya** [comment]
Hey @amyegan  Thanks a lot for the help This worked
